lib/astarte_flow/blocks/device_events_producer/events_decoder.ex

```diff
@@ -123,11 +123,12 @@ defmodule Astarte.Flow.Blocks.DeviceEventsProducer.EventsDecoder do
       v when is_binary(v) ->
         {:ok, :string}
 
-      {subtype, bin} when is_integer(subtype) and is_binary(bin) ->
+      # From Cyanide 2.0, binaries are decoded as %Cyanide.Binary{}
+      %Cyanide.Binary{subtype: subtype, data: bin} when is_atom(subtype) and is_binary(bin) ->
         {:ok, :binary}
 
       v when is_map(v) ->
-        Enum.reduce_while(v, {:ok, %{}}, fn {item_key, item_value}, {:ok, acc} ->
+        Enum.reduce_while(Map.to_list(v), {:ok, %{}}, fn {item_key, item_value}, {:ok, acc} ->
```

Comment:
Nit: maps implement the `Enumerable` protocol, so there is no need to use `to_list/1`